### PR TITLE
Add MKS VNC proxy for VMware

### PR DESCRIPTION
**Pull request by grandchild**
Original code can be found at: https://opendev.org/x/nova-mksproxy

Integrate the code into nova as a builtin command.

Subclass `NovaProxyRequestHandler` for the `WebSocketProxy` by overriding just the handshake part. This made it necessary to refactor the handshake & connection code out of `new_websocket_client()` into a new `handshake_and_connect()` method in `NovaProxyRequestHandler`.

### CODE REVIEW

**Comment by fwiesel** [comment]
I think, I would structure it a bit differently:

Most of the code in `handshake_and_connect` belongs in my view to instance derived of [SecurityProxy](https://github.com/sapcc/nova/blob/stable/rocky-m3/nova/console/securityproxy/base.py), as it is more or less doing the credential handshake.

I would suggest to split out [this part](https://github.com/sapcc/nova/blob/stable/rocky-m3/nova/console/websocketproxy.py#L265-L278) as `process_internal_access_path` (or similar).
Then you can parse the `internal_access_path` and store the necessary values in `self`, which is then passed on to
a `SecurityProxy` in connect as `tenant_sock.reqhandler`.

I admit, it is a bit more convoluted, but I think, it closer matches the purposes of the request-handler and security-proxy classes.

**Comment by grandchild** [comment]
That makes sense, I'll try and rework it like you say.

File: nova/conf/mks.py [hunk 1]
```diff
@@ -13,6 +13,7 @@
 #    License for the specific language governing permissions and limitations
 #    under the License.
 
+import os
```
Comment (fwiesel) [inline nova/conf/mks.py:16]: Is that import needed? I am not seeing a reference to it in the diff

Comment (grandchild) [reply nova/conf/mks.py]: Forgot to run flake8, sorry about that. :/

**Review by fwiesel — APPROVED** [review]

**Comment by grandchild** [comment]
Unit tests found a bug 🎉

**Pull request merged**